packages/mask/src/plugins/NFTCard/SNSAdaptor/NFTCardDialogUI.tsx

```diff
@@ -0,0 +1,123 @@
+import { useRemoteControlledDialog } from '@masknet/shared-base-ui'
+import { Button, Typography } from '@mui/material'
+import { PluginWalletStatusBar, useI18N as useBaseI18n } from '../../../utils'
+import { PluginId } from '@masknet/plugin-infra'
+import { useCurrentWeb3NetworkPluginID } from '@masknet/plugin-infra/web3'
+import { WalletMessages } from '../../../plugins/Wallet/messages'
+import {
+    NetworkPluginID,
+    SourceType,
+    NonFungibleTokenOrder,
+    NonFungibleTokenEvent,
+    Pageable,
+} from '@masknet/web3-shared-base'
+import { NFTCardDialogTabs } from './NFTCardDialog'
+import { useStyles } from '../useStyles'
+import { chainResolver, NetworkType } from '@masknet/web3-shared-evm'
+import { useActivatedPlugin } from '@masknet/plugin-infra/dom'
+import { AboutTab } from './Tabs/AboutTab'
+import { OffersTab } from './Tabs/OffersTab'
+import { ActivityTab } from './Tabs/ActivityTab'
+import { NFTBasicInfo } from '../../../components/shared/NFTCard/NFTBasicInfo'
+import type { ChainId, SchemaType } from '@masknet/web3-shared-evm'
+import type { Web3Helper } from '@masknet/plugin-infra/src/web3-helpers'
+import type { AsyncStateRetry } from 'react-use/lib/useAsyncRetry'
+import { LoadingBase } from '@masknet/theme'
+interface NFTCardDialogUIProps {
+    currentTab: NFTCardDialogTabs
+    asset: AsyncStateRetry<Web3Helper.NonFungibleAssetScope<void, NetworkPluginID.PLUGIN_EVM>>
+    onChangeProvider: (v: SourceType) => void
+    orders: AsyncStateRetry<Pageable<NonFungibleTokenOrder<ChainId, SchemaType>>>
+    events: AsyncStateRetry<Pageable<NonFungibleTokenEvent<ChainId, SchemaType>>>
+    provider: SourceType
+}
+
+const supportedProvider = [
+    // to add providers, temp hide some providers
+    SourceType.OpenSea,
+    SourceType.Gem,
+    SourceType.Rarible,
+    SourceType.NFTScan,
+    // SourceType.X2Y2,
+    // SourceType.Zora,
+    // SourceType.LooksRare,
+]
+
+export function NFTCardDialogUI(props: NFTCardDialogUIProps) {
+    const { currentTab, asset, orders, onChangeProvider, provider, events } = props
+    const { classes } = useStyles()
+    const { t: tb } = useBaseI18n()
+    const { setDialog: setSelectProviderDialog } = useRemoteControlledDialog(
+        WalletMessages.events.selectProviderDialogUpdated,
+    )
+    const pluginId = useCurrentWeb3NetworkPluginID()
+    const chainIdList =
+        useActivatedPlugin(PluginId.NFTCard, 'any')?.enableRequirement.web3?.[NetworkPluginID.PLUGIN_EVM]
+            ?.supportedChainIds ?? []
```

Comment:
Maybe we should fallback to full list of EVM chain ids.
```suggestion
    const chainIdList = pluginDefinition.enableRequirement.web3?.[NetworkPluginID.PLUGIN_EVM].supportedChainIds ?? []
```

@Jack-Works Shall we create a React context for each plugin and wrap components (of course, the root ones) of each plugin inside? So that we can export API as hooks for plugin. For example

- `usePluginConfig()`
- `usePluginSupportedChainIds()`, if `pluginDefinition.enableRequirement.web3` is not provided, it should return the full list of chain ids.
- some other stuffs that we don't want plugins to touch the plugin system internal stuff.